File: modeler/src/main/java/net/sayaya/document/modeler/sample/SampleToDTO.java
package net.sayaya.document.modeler.sample;

import lombok.experimental.UtilityClass;
import net.sayaya.document.data.Sample;

@UtilityClass
public class SampleToDTO {
	public Sample map(net.sayaya.document.modeler.sample.Sample entity) {
		return new Sample().id(entity.id().toString())
				.model(entity.model())
				.name(entity.name())
				.size(entity.size())
				.page(entity.page());
	}
}

File: modeler/src/main/kotlin/net/sayaya/document/modeler/sample/SampleToDTO.kt
package net.sayaya.document.modeler.sample

import org.mapstruct.Mapper
import org.mapstruct.NullValueMappingStrategy
import org.mapstruct.ReportingPolicy
import java.util.*

@Mapper(componentModel = "spring", unmappedTargetPolicy = ReportingPolicy.IGNORE, nullValueMappingStrategy=NullValueMappingStrategy.RETURN_DEFAULT)
interface SampleToDTO {
    fun toSample(entity: Sample): net.sayaya.document.data.Sample
    @JvmDefault fun map(id: UUID): String {
        return id.toString()
    }
}